packages/gasket-plugin-command/lib/index.d.ts

```diff
@@ -0,0 +1,107 @@
+import type { MaybeAsync, GasketConfig } from '@gasket/core';
+import type { PluginData } from '@gasket/plugin-metadata';
+import type { Command } from 'commander';
+
+export interface GasketCommandDefinition {
+  /* Command id/name */
+  id: string;
+
+  /* Command description */
+  description: string;
+
+  /* Command arguments */
+  args?: Array<GasketArgDefinition>;
+
+  /* Command options */
+  options?: Array<GasketOptionDefinition>;
+
+  /* Command action handler */
+  action: (...args: any[]) => MaybeAsync<void>;
+
+  /* Hide from help output */
+  hidden?: boolean;
+
+  /* Default command to run if no command is provided */
+  default?: boolean;
+}
+
```

Comment:
Had a thought a while ago. You know how in the [Validating lifecycle scripts](https://github.com/godaddy/gasket/blob/main/docs/typescript.md#validating-lifecycle-scripts) document how you have to deal with annoyances of JSDocs syntax in order to use generics? Maybe it'd be a good pattern for plugins to export an easy-to-use type for hooks.

```typescript
// add Hook to import from `@gasket/core`

export type CommandHook = Hook<'command'>;
```

That way someone could slap this on their hook for validation:

```javascript
/**
* @type {import('@gasket/plugin-command').CommandHook}
 */
 configure(gasket) {
   // ...
 }
 ```

Reply:
So it would look like:

```js
export default {
  name: 'my-custom-command-plugin',
  hooks: {
    /* @type {import('@gasket/plugin-command').CommandHook} */
    commands() {
      return {
        // custom command
      }
    }
  }
}
```
?

Reply:
I've added the type export and a doc section in the [README](https://github.com/godaddy/gasket/tree/PFX-616/packages/gasket-plugin-command#example-with-jsdoc-types)